RIOT-OS/RIOT
# drivers/ethos: enable multiple instances

### Contribution description

Make `ethos_params_t` and compile-time const array that can have multiple entries for ethos instaces. 


### Testing procedure

`examples/gnrc_border_router` should still work as before.
But now it's also possible to configure multiple `ethos` interfaces:

```C
#define ETHOS_PARAMS            { .uart = UART_DEV(0),   \
                                  .baudrate = 115200 }, \
                                { .uart = UART_DEV(1),   \
                                  .baudrate = 115200 }
```

```
2021-06-11 12:00:37,114 # Iface  7  HWaddr: 5A:84:3B:4F:A8:03 
2021-06-11 12:00:37,115 #           L2-PDU:1500  MTU:1500  HL:64  RTR  
2021-06-11 12:00:37,116 #           RTR_ADV  
2021-06-11 12:00:37,117 #           Source address length: 6
2021-06-11 12:00:37,117 #           Link type: wired
2021-06-11 12:00:37,119 #           inet6 addr: fe80::5884:3bff:fe4f:a803  scope: link  VAL
2021-06-11 12:00:37,120 #           inet6 group: ff02::2
2021-06-11 12:00:37,121 #           inet6 group: ff02::1
2021-06-11 12:00:37,122 #           inet6 group: ff02::1:ff4f:a803
2021-06-11 12:00:37,122 #           
2021-06-11 12:00:37,123 #           Statistics for Layer 2
2021-06-11 12:00:37,123 #             RX packets 0  bytes 0
2021-06-11 12:00:37,125 #             TX packets 1 (Multicast: 1)  bytes 78
2021-06-11 12:00:37,126 #             TX succeeded 0 errors 0
2021-06-11 12:00:37,126 #           Statistics for IPv6
2021-06-11 12:00:37,127 #             RX packets 0  bytes 0
2021-06-11 12:00:37,129 #             TX packets 1 (Multicast: 1)  bytes 64
2021-06-11 12:00:37,130 #             TX succeeded 1 errors 0
2021-06-11 12:00:37,130 # 
2021-06-11 12:00:37,131 # Iface  6  HWaddr: 5A:84:3B:4F:A8:02 
2021-06-11 12:00:37,133 #           L2-PDU:1500  MTU:1500  HL:64  RTR  
2021-06-11 12:00:37,133 #           RTR_ADV  
2021-06-11 12:00:37,134 #           Source address length: 6
2021-06-11 12:00:37,135 #           Link type: wired
2021-06-11 12:00:37,137 #           inet6 addr: fe80::5884:3bff:fe4f:a802  scope: link  VAL
2021-06-11 12:00:37,139 #           inet6 group: ff02::2
2021-06-11 12:00:37,140 #           inet6 group: ff02::1
2021-06-11 12:00:37,141 #           inet6 group: ff02::1:ff4f:a802
2021-06-11 12:00:37,141 #           
2021-06-11 12:00:37,142 #           Statistics for Layer 2
2021-06-11 12:00:37,143 #             RX packets 0  bytes 0
2021-06-11 12:00:37,145 #             TX packets 1 (Multicast: 1)  bytes 78
2021-06-11 12:00:37,146 #             TX succeeded 0 errors 0
2021-06-11 12:00:37,148 #           Statistics for IPv6
2021-06-11 12:00:37,149 #             RX packets 0  bytes 0
2021-06-11 12:00:37,150 #             TX packets 1 (Multicast: 1)  bytes 64
2021-06-11 12:00:37,151 #             TX succeeded 1 errors 0
2021-06-11 12:00:37,152 # 

2021-06-11 12:02:47,759 # ps
2021-06-11 12:02:47,761 # 	pid | name                 | state    Q | pri | stack  ( used) ( free) | base addr  | current     
2021-06-11 12:02:47,764 # 	  - | isr_stack            | -        - |   - |    512 (  232) (  280) | 0x20000000 | 0x200001c8
2021-06-11 12:02:47,766 # 	  1 | main                 | running  Q |   7 |   1536 ( 1136) (  400) | 0x20000864 | 0x20000b04 
2021-06-11 12:02:47,769 # 	  2 | pktdump              | bl rx    _ |   6 |   1536 (  240) ( 1296) | 0x20004a34 | 0x20004f44 
2021-06-11 12:02:47,770 # 	  3 | ipv6                 | bl rx    _ |   4 |   1024 (  580) (  444) | 0x20000f40 | 0x2000123c 
2021-06-11 12:02:47,773 # 	  4 | udp                  | bl rx    _ |   5 |   1024 (  256) (  768) | 0x20005694 | 0x20005994 
2021-06-11 12:02:47,775 # 	  5 | usbus                | bl anyfl _ |   1 |   1024 (  392) (  632) | 0x200002f8 | 0x200005fc 
2021-06-11 12:02:47,778 # 	  6 | ethos                | bl rx    _ |   2 |   1024 (  588) (  436) | 0x20002710 | 0x200029cc 
2021-06-11 12:02:47,781 # 	  7 | ethos                | bl rx    _ |   2 |   1024 (  588) (  436) | 0x20002b10 | 0x20002dcc 
2021-06-11 12:02:47,783 # 	    | SUM                  |            |     |   8704 ( 4012) ( 4692)

2021-06-11 12:02:50,463 # ping ff02::1
2021-06-11 12:02:50,478 # 12 bytes from fe80::a4ed:29ff:feec:d5f3%7: icmp_seq=0 ttl=64 time=13.176 ms
2021-06-11 12:02:50,485 # 12 bytes from fe80::ece6:c0ff:fe26:b89e%6: icmp_seq=0 ttl=64 time=19.839 ms (DUP!)
2021-06-11 12:02:51,478 # 12 bytes from fe80::a4ed:29ff:feec:d5f3%7: icmp_seq=1 ttl=64 time=13.168 ms
2021-06-11 12:02:51,485 # 12 bytes from fe80::ece6:c0ff:fe26:b89e%6: icmp_seq=1 ttl=64 time=19.822 ms (DUP!)
2021-06-11 12:02:52,478 # 12 bytes from fe80::a4ed:29ff:feec:d5f3%7: icmp_seq=2 ttl=64 time=13.168 ms
2021-06-11 12:02:52,478 # 
2021-06-11 12:02:52,479 # --- ff02::1 PING statistics ---
2021-06-11 12:02:52,481 # 3 packets transmitted, 3 packets received, 2 duplicates, 0% packet loss
2021-06-11 12:02:52,482 # round-trip min/avg/max = 13.168/15.834/19.839 ms
```


### Issues/PRs references
